js/components/PortfolioTable.jsx

```diff
@@ -16,9 +16,10 @@ var PortfolioTable = React.createClass({
 
   getPortfolioItems: function(privateFields) {
     var portfolioItems = [];
-    var length = this.props.user.portfolio.length;
-    for (var i = 0; i < length; i++) {
-      var category = this.props.user.portfolio[i];
+    var portfolio = this.props.user.portfolio;
+    var sortedPortfolio = this.props.user.portfolio.sort(this.getPortfolioComparator());
```

Comment:
Same thing here. Calling sort on `this.props.user.portfolio` will sort in place and return the new Array.